src/lib.rs

```diff
@@ -2,67 +2,144 @@ use pyo3::{
+    let directory = path.parent().unwrap();
+    println!("This is the file name {:?}", directory);
+
+    let mut filenames = Vec::new();
+
+    for entry in WalkDir::new(directory)
+        .into_iter()
+        .filter_map(Result::ok)
+        .filter(|e| !e.file_type().is_dir())
+    {
+        let f_name = String::from(entry.file_name().to_string_lossy());
+        let absoltue_name = String::from(entry.path().to_string_lossy());
 
-#[pymethods]
-impl DomNode {
-    #[new]
-    pub fn new(
-        tag: String,
-        props: Py<PyDict>,
-        children: Py<PyList>,
-        event_listeners: Py<PyDict>,
-        state: Py<PyDict>,
-    ) -> DomNode {
-        DomNode {
-            tag,
-            props,
-            children,
-            event_listeners,
-            state,
-            original_data: "".to_string(),
-            data: "".to_string(),
-            parent_element: None,
-            dom_pointer: None,
+        if f_name.ends_with(".fyre") {
+            filenames.push(absoltue_name);
         }
     }
 
-    pub fn is_text_node(&self) -> bool {
-        self.tag == "TEXT_NODE"
-    }
+    filenames
+}
+
+fn get_function_template(pyml_lines: &str, file_name: &str, root_name: &str) -> String {
+    let file_name = file_name.replace(".py", "");
+
+    if file_name.contains("__init__") {
+        return format!(
+            "
+from starfyre import create_component, render
+
+def fx_{0}():
+    # not nesting the code to preserve the frames
+    component = create_component(\"\"\"
+{1}
+        \"\"\")
+
+    return render(component)
+
+
+{0}=fx_{0}()
+    ",
+            root_name, pyml_lines
+        );
+    } else {
+        return format!(
+            "
+from starfyre import create_component
+
+def fx_{0}():
+    component = create_component(\"\"\"
+{1}
+        \"\"\")
+
+    return component
 
-    pub fn set_parent_element(&mut self, parent_element: Py<PyAny>) {
-        self.parent_element = Some(parent_element);
+
+{0}=fx_{0}()
+    ",
+            root_name, pyml_lines
+        );
     }
 }
 
-pub struct DomTree {
-    pub root: DomNode,
-}
+#[pyfunction]
+fn compile(file_name: &str) {
+    // get all the files in the directory of the file that end with .fyre
+    let fyre_files = get_all_fyre_files(file_name);
+
+    for fyre_file in fyre_files {
+        let python_file = fyre_file.replace(".fyre", ".py");
+
+        let fyre_file_pointer = File::open(fyre_file).unwrap();
+        let fyre_file_reader = std::io::BufReader::new(fyre_file_pointer);
+        let mut python_lines: Vec<String> = vec![];
+        let mut pyml_lines = vec![];
+
+        let mut python_flag = true;
+        let mut root_name = String::from("");
+
+        for line in fyre_file_reader.lines() {
+            let copy_line = line.unwrap().clone();
+            let is_pyml_line = copy_line.clone().trim().starts_with("<");
+            if !python_flag || is_pyml_line {
+                python_flag = false;
```

Comment:
```suggestion
            if is_pyml_line {
```
Would this also work? and doing `if/else` (or `if` early exit) instead of `if/else if `
May make this easier to reason about

Reply:
Yep. early exit would be objectively cleaner code.